### ISSUE TITLE
WiFi card not detected - Broadcom BCM943228HMB

### ISSUE THREAD

**Issue by wesleylancel** [home-assistant/operating-system#2525]
### Describe the issue you are experiencing

The built-in WiFi card of my HP T620 thin client - a Broadcom BCM943228HMB - is not being recognized.

### What operating system image do you use?

generic-x86-64 (Generic UEFI capable x86-64 systems)

### What version of Home Assistant Operating System is installed?

10.1

### Did you upgrade the Operating System.

Yes

### Steps to reproduce the issue

1. Install Home Assistant generic image
2. Check if WiFi works (verified through `nmcli device wifi`)

### Anything in the Supervisor logs that might be useful for us?

```txt
-
```

### Anything in the Host logs that might be useful for us?

```txt
Apr 30 19:44:33 homeassistant kernel: audit: type=1300 audit(1682883873.076:168): arch=c000003e syscall=46 success=yes exit=508 a0=3 a1=7ffc4b7868e0 a2=0 a3=7ffc4b7868cc items=0 ppid=2029 pid=2047 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="iptables" exe="/usr/sbin/xtables-nft-multi" subj=unconfined key=(null)
Apr 30 19:44:33 homeassistant kernel: audit: type=1327 audit(1682883873.076:168): proctitle=2F7573722F7362696E2F69707461626C6573002D2D77616974002D74006E6174002D4900444F434B45525F504F5354524F5554494E47002D73003132372E302E302E3131002D7000746370002D2D73706F7274003431393639002D6A00534E4154002D2D746F2D736F75726365003A3533
Apr 30 19:44:33 homeassistant kernel: eth0: renamed from vetha75d35d
Apr 30 19:44:33 homeassistant kernel: IPv6: ADDRCONF(NETDEV_CHANGE): veth3ed27ec: link becomes ready
Apr 30 19:44:33 homeassistant kernel: hassio: port 5(veth3ed27ec) entered blocking state
Apr 30 19:44:33 homeassistant kernel: hassio: port 5(veth3ed27ec) entered forwarding state
Apr 30 19:44:33 homeassistant NetworkManager[377]: <info>  [1682883873.1036] device (veth3ed27ec): carrier: link connected
Apr 30 19:44:33 homeassistant systemd-resolved[372]: System hostname changed to 'homeassistant'.
Apr 30 19:44:33 homeassistant systemd-resolved[372]: Detected conflict on homeassistant.local IN A 192.168.123.98
Apr 30 19:44:33 homeassistant systemd-resolved[372]: Hostname conflict, changing published hostname from 'homeassistant' to 'homeassistant10'.
Apr 30 19:44:33 homeassistant systemd[1]: Started libcontainer container d05fcd6fc20d7fffb5af332cc65e0e098e4b61d9fbba2cc2d1d04ec62f8b7f3e.
Apr 30 19:44:33 homeassistant kernel: audit: type=1334 audit(1682883873.968:169): prog-id=38 op=LOAD
Apr 30 19:44:33 homeassistant systemd-resolved[372]: System hostname changed to 'homeassistant'.
Apr 30 19:44:34 homeassistant systemd-resolved[372]: Detected conflict on homeassistant.local IN A 192.168.123.98
Apr 30 19:44:34 homeassistant systemd-resolved[372]: Hostname conflict, changing published hostname from 'homeassistant' to 'homeassistant2'.
Apr 30 19:44:35 homeassistant bluetoothd[378]: Endpoint registered: sender=:1.36 path=/MediaEndpoint/A2DPSink/sbc
Apr 30 19:44:35 homeassistant bluetoothd[378]: Endpoint registered: sender=:1.36 path=/MediaEndpoint/A2DPSource/sbc
Apr 30 19:44:35 homeassistant kernel: Bluetooth: RFCOMM TTY layer initialized
Apr 30 19:44:35 homeassistant kernel: Bluetooth: RFCOMM socket layer initialized
Apr 30 19:44:35 homeassistant kernel: Bluetooth: RFCOMM ver 1.11
Apr 30 19:44:36 homeassistant systemd[1]: var-lib-docker-overlay2-b9659d7c2a2bfe02a63c06f8f132fd7dc4e63e81fdbb7f2b4b45f30886c57b0b\x2dinit-merged.mount: Deactivated successfully.
Apr 30 19:44:36 homeassistant systemd[1]: mnt-data-docker-overlay2-b9659d7c2a2bfe02a63c06f8f132fd7dc4e63e81fdbb7f2b4b45f30886c57b0b\x2dinit-merged.mount: Deactivated successfully.
Apr 30 19:44:37 homeassistant systemd[1]: Started libcontainer container ed84dede8ed1ec37b6953ceea6a3548e122a63d5180ad79fea00342f203c523b.
Apr 30 19:44:37 homeassistant systemd-resolved[372]: System hostname changed to 'homeassistant'.
Apr 30 19:44:37 homeassistant systemd-resolved[372]: Detected conflict on homeassistant IN A 192.168.123.98
Apr 30 19:44:37 homeassistant systemd-resolved[372]: Hostname conflict, changing published hostname from 'homeassistant' to 'homeassistant4'.
Apr 30 19:44:39 homeassistant systemd[1]: NetworkManager-dispatcher.service: Deactivated successfully.
Apr 30 19:44:42 homeassistant systemd[1]: Started libcontainer container cc42030f66648ca14c57f5556b0774d9119ee79ba57b052f3606e07b7d9edb01.
Apr 30 19:44:42 homeassistant kernel: kauditd_printk_skb: 23 callbacks suppressed
Apr 30 19:44:42 homeassistant kernel: audit: type=1334 audit(1682883882.592:181): prog-id=46 op=LOAD
Apr 30 19:44:42 homeassistant kernel: audit: type=1300 audit(1682883882.592:181): arch=c000003e syscall=321 success=yes exit=14 a0=5 a1=c0001377f8 a2=78 a3=0 items=0 ppid=3032 pid=3044 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="runc" exe="/usr/bin/runc" subj=unconfined key=(null)
Apr 30 19:44:42 homeassistant kernel: audit: type=1327 audit(1682883882.592:181): proctitle=72756E63002D2D726F6F74002F7661722F72756E2F646F636B65722F72756E74696D652D72756E632F6D6F6279002D2D6C6F67002F72756E2F636F6E7461696E6572642F696F2E636F6E7461696E6572642E72756E74696D652E76322E7461736B2F6D6F62792F63633432303330663636363438636131346335376635353536
Apr 30 19:44:42 homeassistant kernel: audit: type=1334 audit(1682883882.593:182): prog-id=47 op=LOAD
Apr 30 19:44:42 homeassistant kernel: audit: type=1300 audit(1682883882.593:182): arch=c000003e syscall=321 success=yes exit=16 a0=5 a1=c000137590 a2=78 a3=0 items=0 ppid=3032 pid=3044 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="runc" exe="/usr/bin/runc" subj=unconfined key=(null)
Apr 30 19:44:42 homeassistant kernel: audit: type=1327 audit(1682883882.593:182): proctitle=72756E63002D2D726F6F74002F7661722F72756E2F646F636B65722F72756E74696D652D72756E632F6D6F6279002D2D6C6F67002F72756E2F636F6E7461696E6572642F696F2E636F6E7461696E6572642E72756E74696D652E76322E7461736B2F6D6F62792F63633432303330663636363438636131346335376635353536
Apr 30 19:44:42 homeassistant kernel: audit: type=1334 audit(1682883882.593:183): prog-id=47 op=UNLOAD
Apr 30 19:44:42 homeassistant kernel: audit: type=1334 audit(1682883882.593:184): prog-id=46 op=UNLOAD
Apr 30 19:44:42 homeassistant kernel: audit: type=1334 audit(1682883882.593:185): prog-id=48 op=LOAD
Apr 30 19:44:42 homeassistant kernel: audit: type=1300 audit(1682883882.593:185): arch=c000003e syscall=321 success=yes exit=14 a0=5 a1=c000137a50 a2=78 a3=0 items=0 ppid=3032 pid=3044 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="runc" exe="/usr/bin/runc" subj=unconfined key=(null)
Apr 30 19:44:42 homeassistant systemd-resolved[372]: System hostname changed to 'homeassistant'.
Apr 30 19:44:42 homeassistant systemd-resolved[372]: Detected conflict on homeassistant.local IN A 192.168.123.98
Apr 30 19:44:42 homeassistant systemd-resolved[372]: Hostname conflict, changing published hostname from 'homeassistant' to 'homeassistant11'.
Apr 30 19:44:58 homeassistant systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Apr 30 19:44:58 homeassistant kernel: kauditd_printk_skb: 1 callbacks suppressed
Apr 30 19:44:58 homeassistant kernel: audit: type=1334 audit(1682883898.139:186): prog-id=14 op=UNLOAD
Apr 30 19:44:58 homeassistant kernel: audit: type=1334 audit(1682883898.139:187): prog-id=13 op=UNLOAD
Apr 30 19:44:58 homeassistant kernel: audit: type=1334 audit(1682883898.139:188): prog-id=12 op=UNLOAD
Apr 30 19:44:58 homeassistant systemd[1]: systemd-timedated.service: Deactivated successfully.
Apr 30 19:44:58 homeassistant kernel: audit: type=1334 audit(1682883898.451:189): prog-id=25 op=UNLOAD
Apr 30 19:44:58 homeassistant kernel: audit: type=1334 audit(1682883898.451:190): prog-id=24 op=UNLOAD
Apr 30 19:44:58 homeassistant kernel: audit: type=1334 audit(1682883898.451:191): prog-id=23 op=UNLOAD
Apr 30 19:45:09 homeassistant systemd[1]: var-lib-docker-overlay2-f9dfd393a8b293963f442af3c3680753fa543e4253ac8d04be177a2f526ce642\x2dinit-merged.mount: Deactivated successfully.
Apr 30 19:45:09 homeassistant systemd[1]: mnt-data-docker-overlay2-f9dfd393a8b293963f442af3c3680753fa543e4253ac8d04be177a2f526ce642\x2dinit-merged.mount: Deactivated successfully.
Apr 30 19:45:09 homeassistant kernel: hassio: port 6(vethdeab500) entered blocking state
Apr 30 19:45:09 homeassistant kernel: hassio: port 6(vethdeab500) entered disabled state
Apr 30 19:45:09 homeassistant kernel: device vethdeab500 entered promiscuous mode
Apr 30 19:45:09 homeassistant kernel: audit: type=1700 audit(1682883909.332:192): dev=vethdeab500 prom=256 old_prom=0 auid=4294967295 uid=0 gid=0 ses=4294967295
Apr 30 19:45:09 homeassistant kernel: audit: type=1300 audit(1682883909.332:192): arch=c000003e syscall=44 success=yes exit=40 a0=c a1=c000eedc20 a2=28 a3=0 items=0 ppid=1 pid=463 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="dockerd" exe="/usr/bin/dockerd" subj=unconfined key=(null)
Apr 30 19:45:09 homeassistant kernel: audit: type=1327 audit(1682883909.332:192): proctitle=2F7573722F62696E2F646F636B657264002D480066643A2F2F002D2D636F6E7461696E6572643D2F72756E2F636F6E7461696E6572642F636F6E7461696E6572642E736F636B
Apr 30 19:45:09 homeassistant NetworkManager[377]: <info>  [1682883909.3374] manager: (veth16c699b): new Veth device (/org/freedesktop/NetworkManager/Devices/17)
Apr 30 19:45:09 homeassistant NetworkManager[377]: <info>  [1682883909.3432] manager: (vethdeab500): new Veth device (/org/freedesktop/NetworkManager/Devices/18)
Apr 30 19:45:09 homeassistant systemd[1]: Started libcontainer container 295f2bce7ce9d7a546295ef7524c1105b61c5d490e354f67638460116edd3bf0.
Apr 30 19:45:09 homeassistant kernel: audit: type=1334 audit(1682883909.530:193): prog-id=49 op=LOAD
Apr 30 19:45:09 homeassistant kernel: audit: type=1334 audit(1682883909.531:194): prog-id=50 op=LOAD
Apr 30 19:45:09 homeassistant kernel: audit: type=1300 audit(1682883909.531:194): arch=c000003e syscall=321 success=yes exit=15 a0=5 a1=c0001357f8 a2=78 a3=0 items=0 ppid=3460 pid=3472 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="runc" exe="/usr/bin/runc" subj=unconfined key=(null)
Apr 30 19:45:09 homeassistant kernel: audit: type=1327 audit(1682883909.531:194): proctitle=72756E63002D2D726F6F74002F7661722F72756E2F646F636B65722F72756E74696D652D72756E632F6D6F6279002D2D6C6F67002F72756E2F636F6E7461696E6572642F696F2E636F6E7461696E6572642E72756E74696D652E76322E7461736B2F6D6F62792F32393566326263653763653964376135343632393565663735
Apr 30 19:45:09 homeassistant kernel: audit: type=1334 audit(1682883909.531:195): prog-id=51 op=LOAD
Apr 30 19:45:09 homeassistant kernel: audit: type=1300 audit(1682883909.531:195): arch=c000003e syscall=321 success=yes exit=17 a0=5 a1=c000135590 a2=78 a3=0 items=0 ppid=3460 pid=3472 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="runc" exe="/usr/bin/runc" subj=unconfined key=(null)
Apr 30 19:45:09 homeassistant kernel: audit: type=1327 audit(1682883909.531:195): proctitle=72756E63002D2D726F6F74002F7661722F72756E2F646F636B65722F72756E74696D652D72756E632F6D6F6279002D2D6C6F67002F72756E2F636F6E7461696E6572642F696F2E636F6E7461696E6572642E72756E74696D652E76322E7461736B2F6D6F62792F32393566326263653763653964376135343632393565663735
Apr 30 19:45:10 homeassistant kernel: eth0: renamed from veth16c699b
Apr 30 19:45:10 homeassistant kernel: IPv6: ADDRCONF(NETDEV_CHANGE): vethdeab500: link becomes ready
Apr 30 19:45:10 homeassistant kernel: hassio: port 6(vethdeab500) entered blocking state
Apr 30 19:45:10 homeassistant kernel: hassio: port 6(vethdeab500) entered forwarding state
Apr 30 19:45:10 homeassistant NetworkManager[377]: <info>  [1682883910.0247] device (vethdeab500): carrier: link connected
Apr 30 19:45:10 homeassistant systemd-resolved[372]: System hostname changed to 'homeassistant'.
Apr 30 19:45:10 homeassistant systemd-resolved[372]: Detected conflict on homeassistant IN A 192.168.123.98
Apr 30 19:45:10 homeassistant systemd-resolved[372]: Hostname conflict, changing published hostname from 'homeassistant' to 'homeassistant4'.
Apr 30 19:45:15 homeassistant kernel: kauditd_printk_skb: 29 callbacks suppressed
Apr 30 19:45:15 homeassistant kernel: audit: type=1334 audit(1682883915.212:207): prog-id=53 op=LOAD
Apr 30 19:45:15 homeassistant kernel: audit: type=1334 audit(1682883915.213:208): prog-id=54 op=LOAD
Apr 30 19:45:15 homeassistant kernel: audit: type=1334 audit(1682883915.213:209): prog-id=55 op=LOAD
Apr 30 19:45:15 homeassistant systemd[1]: Starting Hostname Service...
Apr 30 19:45:15 homeassistant systemd[1]: Started Hostname Service.
Apr 30 19:45:15 homeassistant kernel: audit: type=1334 audit(1682883915.745:210): prog-id=56 op=LOAD
Apr 30 19:45:15 homeassistant kernel: audit: type=1334 audit(1682883915.746:211): prog-id=57 op=LOAD
Apr 30 19:45:15 homeassistant kernel: audit: type=1334 audit(1682883915.746:212): prog-id=58 op=LOAD
Apr 30 19:45:15 homeassistant systemd[1]: Starting Time & Date Service...
Apr 30 19:45:16 homeassistant systemd[1]: Started Time & Date Service.
Apr 30 19:45:45 homeassistant systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Apr 30 19:45:45 homeassistant kernel: audit: type=1334 audit(1682883945.788:213): prog-id=55 op=UNLOAD
Apr 30 19:45:45 homeassistant kernel: audit: type=1334 audit(1682883945.788:214): prog-id=54 op=UNLOAD
Apr 30 19:45:45 homeassistant kernel: audit: type=1334 audit(1682883945.788:215): prog-id=53 op=UNLOAD
Apr 30 19:45:46 homeassistant systemd[1]: systemd-timedated.service: Deactivated successfully.
Apr 30 19:45:46 homeassistant kernel: audit: type=1334 audit(1682883946.301:216): prog-id=58 op=UNLOAD
Apr 30 19:45:46 homeassistant kernel: audit: type=1334 audit(1682883946.301:217): prog-id=57 op=UNLOAD
Apr 30 19:45:46 homeassistant kernel: audit: type=1334 audit(1682883946.301:218): prog-id=56 op=UNLOAD
Apr 30 19:49:43 homeassistant kernel: audit: type=1334 audit(1682884183.817:219): prog-id=59 op=LOAD
Apr 30 19:49:43 homeassistant systemd-timesyncd[453]: Network configuration changed, trying to establish connection.
Apr 30 19:49:43 homeassistant systemd-timesyncd[453]: Contacted time server 162.159.200.1:123 (time.cloudflare.com).
Apr 30 19:49:43 homeassistant systemd[1]: Started Journal Gateway Service.
Apr 30 19:49:44 homeassistant systemd-journal-gatewayd[3927]: microhttpd: MHD_OPTION_EXTERNAL_LOGGER is not the first option specified for the daemon. Some messages may be printed by the standard MHD logger.
```

## System Information

version | core-2023.4.6
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.10.10
os_name | Linux
os_version | 6.1.25
arch | x86_64
timezone | Europe/Brussels
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4970
Installed Version | 1.32.1
Stage | running
Available Repositories | 1275
Downloaded Repositories | 4

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 10.1
-- | --
update_channel | stable
supervisor_version | supervisor-2023.04.1
agent_version | 1.5.1
docker_version | 23.0.3
disk_total | 14.0 GB
disk_used | 5.5 GB
healthy | true
supported | true
board | generic-x86-64
supervisor_api | ok
version_api | ok
installed_addons | Cloudflared (4.1.4), File editor (5.6.0), SSH & Web Terminal (13.1.0)

</details>

<details><summary>Dashboards</summary>

dashboards | 1
-- | --
resources | 0
mode | auto-gen

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 30 april 2023 om 17:58
-- | --
current_recorder_run | 30 april 2023 om 21:44
estimated_db_size | 0.76 MiB
database_engine | sqlite
database_version | 3.38.5

</details>

**Comment by agners** [comment]
The latest nightly build have the driver included, can you give this a try?
https://os-builds.home-assistant.io/11.0.dev20230501/

**Comment by wesleylancel** [comment]
@agners wow, thx for the quick fix! Is there an easy way for me to update my running system to a dev build?

**Comment by agners** [comment]
Yes, you can temporarily switch to the dev channel:

```
ha supervisor options --channel=dev
ha supervisor reload
ha supervisor update
```

Check if there is indeed an OS update an install it
```
ha os info
ha os update
```

After the update, don't forget to go back to the stable channel. The OS won't automatically downgrade, so you can continue using the dev build:
```
ha supervisor options --channel=stable
ha supervisor reload
```

To downgrade to the latest OR release, you have to downgrade manually then:
```
ha os update --version=10.1
```

**Comment by wesleylancel** [comment]
Hi @agners, thanks for the help! I've managed to update but am not getting WiFi to work yet.

`nmcli device wifi` still doesn't show anything, and rescan doesn't work.

`nmcli radio` shows `WIFI` as enabled (not sure if it did that before but `WIFI-HW` as missing`.

**Comment by agners** [comment]
Can you check the host logs if the firmware has been loaded, e.g. 

```
ha host logs -t kernel -n 1000000 -b 0 | grep firmware
```

Also check if the upgrade to the dev version worked using

```
ha os info
```

**Comment by agners** [comment]
This should be resolved with HAOS 10.2.

**Comment by PhilipGarnero** [comment]
Same issue for me on my old macbook pro 2012 A1278.
The wifi card isn't detected. Apparently, it is a common issue on some linux platforms where the b43 driver is missing (in our case, because they removed the non-free folder from the alpine distribution).
It didn't work on 10.1, I just upgraded to 10.2 but it still isn't showing with `nmcli radio`

**Comment by agners** [comment]
@PhilipGarnero so yours is a BCM43* then? Can you open a separate issue?

@ahmetem can you check the host logs if you see firmware errors or similar?

```
ha host logs -t kernel -n 1000000 -b 0 | grep firmware
```

**Comment by PhilipGarnero** [comment]
Hi @agners !
No it's the same family of broadcom BCM94* cards. Looking online, I only found [some](https://wiki.debian.org/MacBook/Wireless) [references](https://gist.github.com/niftylettuce/5619c2be9906bcbd893e1e1a25b9d795) to a missing b43-firmware for Macbooks, so maybe it isn't related after all.
I've run the command you gave and it yielded nothing. I've tried adding bcm or broadcom or wlan or wifi but nothing either.
I skimmed around the full logs but nothing seems to be resulting in an error.
